docs/taiko_in_docker.md

```diff
@@ -2,42 +2,120 @@
 layout: page.njk
 ---
 
-To run Taiko in a docker container follow these instructions.
+## Running Taiko in docker
 
-## Installing
+Sample code used in this documentation is available on Github. 
 
+The tutorial shows how to run Gauge and Taiko tests using Taiko. The docker image created as a part of this tutorial can be used to run tests locally and well or on a CI/CD setup.
 
-    docker pull node
+## Prerequisites
 
-Start a container
+* [NodeJS](https://nodejs.org/en/)
+* [Docker](https://www.docker.com/)
 
-    docker run -it node /bin/bash
+## Setup a sample project
 
-Update packages
+Create a Gauge+Taiko project. This article (and the sample) uses the default Gauge JavaScript template:
 
-    apt-get update
+```
+gauge init js
+```
 
-Install chromium dependencies
+## Create a Docker image
 
-    apt-get install -y wget unzip fontconfig locales gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils
+Use the [docker file](https://docs.docker.com/engine/reference/builder/) listed below to create a reusable docker image.
 
-Install Taiko
+## Dockerfile
 
-    npm install taiko -g --allow-root --unsafe-perm=true
+```
+# This image uses the official node base image.
+FROM node:12.18.3-buster-slim@sha256:dd6aa3ed10af4374b88f8a6624aeee7522772bb08e8dd5e917ff729d1d3c3a4f
+ 
+# The Taiko installation downloads and installs the chromium required to run the tests. 
+# However, we need the chromium dependencies installed in the environment. These days, most # Dockerfiles just install chrome to get the dependencies.
+RUN apt-get update \
+     && apt-get install -y wget gnupg ca-certificates \
+     && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
+     && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
+     && apt-get update \
+     && apt-get install -y google-chrome-stable
+  
+# Set a custom npm install location so that Gauge, Taiko and dependencies can be 
+# installed without root privileges
+ENV NPM_CONFIG_PREFIX=/home/gauge/.npm-packages
+ENV PATH="${NPM_CONFIG_PREFIX}/bin:${PATH}"
 
-Note: These instructions should ideally be a part of the `Dockerfile` while
-building a docker image.
+# Add the Taiko browser arguments
+ENV TAIKO_BROWSER_ARGS=--no-sandbox,--start-maximized,--disable-dev-shm-usage
+ENV headless_chrome=true
```

Comment:
Is this used in test?

Reply:
Yes these are used by Taiko. This is to keep the command to invoke Taiko in the command line simple.

Reply:
Yes, these are used by Taiko. Since this is in the docker file, this keeps the command to invoke tests in the command line simple.

Reply:
I'm looking at keeping a generic Dockerfile and hold nothing with regards to test.

Reply:
This file is a copy of something that can run for `gauge init js` template. Which is at https://github.com/getgauge/template-js/blob/0cbaca7e8f8b402e44c0d490793292e3d406fc26/tests/step_implementation.js#L9 

But I agree that it should be as clean as possible. Will see how I can modify the template to avoid adding this in the docker file.

Reply:
@sguptatw for now please remove all content and link to the docker file at https://github.com/getgauge/template-js/blob/master/Dockerfile and write instructions where to find this file after a `gauge init js`